src/ui/settingsPage/settingsPage.js

```diff
@@ -0,0 +1,186 @@
+          <!-- User has no color preference: -->
+          <img src="../../assets/logos/logo-dark.svg" />
+        </picture>
+        <h1>Mozilla VPN</h1>
+      </header>
+      <main>
+        <h2>Website Preferences</h2>
+        <p>
+          You can manage VPN preferences for each website in the Mozilla VPN for
+          Firefox extension.
+        </p>
+        <div>
+          <img />
+          <input
+            type="text"
+            placeholder="Search Websites"
+            ${ref(this.filterInput)}
+            @change=${() => this.requestUpdate()}
+            @input=${() => this.requestUpdate()}
+          />
+        </div>
+        <mz-context-table
+          .contexts=${filteredList}
+          .serverList=${this.serverList}
+          .onRemoveOrigin=${this.removeOrigin.bind(this)}
+        ></mz-context-table>
+      </main>
+    `;
+  }
+  removeOrigin(origin) {
+    this.proxyHandler.then((handler) => {
+      handler.removeContextForOrigin(origin);
+    });
+  }
+
+  static styles = css`
+    ${fontSizing}
+    ${resetSizing}
+
+  body {
+      width: var(--window-width);
+      background-color: var(--panel-bg-color);
+      padding: 0;
+      margin: 0;
+    }
+
+    :host {
+      background-color: var(--panel-bg-color);
+      display: block;
+      width: 100%;
+      height: 100vh;
+      margin: 0;
+      padding: 0;
+      display: flex;
+      flex-direction: column;
+      align-items: center;
+    }
+    * {
+      color: var(--text-color-primary);
+    }
+    header {
+      display: flex;
+      flex-direction: row;
+      padding: 20px 30px;
+      border-bottom: 1px solid var(--border-color);
+      width: 100%;
+      margin-bottom: 16px;
+    }
+    main {
+      width: calc(min(70%, 920px));
+    }
+
+    picture img {
+      margin: 0px 5px;
+      height: 30px;
+      width: 30px;
+    }
+
+    h2 {
+      font-size: 38px;
+      font-style: normal;
+      font-weight: 700;
+      line-height: 40px; /* 105.263% */
+      margin-bottom: 18px;
+    }
+    p {
+      font-size: 16px;
+      font-style: normal;
+      font-weight: 400;
+      line-height: 169.336%;
+      margin-bottom: 32px;
+    }
+    input {
+      margin-bottom: 32px;
+      padding: 10px 20px;
+      padding-left: 30px;
+      color: var(--text-color-invert);
+      width: calc(max(50%, 300px));
+      background-image: url("../../assets/img/search-icon.svg");
+      background-position: 2.5px 6px;
+      background-repeat: no-repeat;
+      border: 2px solid var(--border-color);
+      border-radius: 5px;
+    }
+
+    main p {
+      font-size: 14px;
+      font-style: normal;
+      font-weight: 400;
+      line-height: 21px;
+    }
+  `;
+}
+customElements.define("mz-settingspage", SettingsPage);
+
+/**
+ * Takes a map of sitecontexts
+ * @param {Map<String,SiteContext>} siteContextList
+ */
+export const filter = (siteContextList, filterString = "") => {
+  const out = [];
+  siteContextList.keys().forEach((key) => {
```

Comment:
When running this branch locally I get a `siteContextList.keys().forEach... is not a function` error here and the list doesn't show. Changing this to `for (const [key] of siteContextList.entries())` fixes it.

Reply:
wierd